compat/src/index.d.ts

```diff
@@ -0,0 +1,71 @@
+import * as hooks from '../../hooks';
+import { VNode, PreactElement } from '../../src/internal';
+import { Component, createContext, createRef, Fragment, toChildArray, ComponentChildren, ComponentChild } from '../../src';
+import { FunctionalComponent } from 'preact';
+import { ForwardFn } from './internal';
+
+export { Component, createContext, createRef, Fragment } from '../../src';
+export * from '../../hooks';
+
+export const version: string;
+
+export function createPortal (vnode: VNode, container: PreactElement): VNode<any>;
+
+export function createElement (type: VNode["type"], props?: any, Children?: ComponentChildren): VNode<any>;
+
+export function cloneElement (element: VNode<any>): VNode<any>;
```

Comment:
Maybe we can reuse the type from `preact`?

Reply:
I moved things all into a `compat` namespace so they can be used via default import as well. How do I add imported types to an exported namespace? `import export` worked for hooks as it was already a namespace.

Reply:
Got it figured out! This has been completed